### PR TITLE
[10.x] Fix route:cache from excluding octane routes

### PR DESCRIPTION
**Pull request by CharlesBilbo**
When running route:cache command with octane routes they are being excluded from the cached routes file. This causes octane routes to 404 when cached.

While caching the routes gives a negative performance to swoole (not sure about road runner) by increasing the amount of memory needed for each request this will at least prevent them prevent them from 404ing.

This will also prevent questions like this

https://github.com/laravel/octane/issues/596
https://github.com/laravel/octane/issues/658

**This PR is a required dependency for this PR.** 

https://github.com/laravel/octane/pull/659

Additonally it doesnt seem like Octane::route method doesn't work with road runner.

### CODE REVIEW

File: src/Illuminate/Foundation/Console/RouteCacheCommand.php [hunk 1]
```diff
@@ -78,10 +79,33 @@ public function handle()
...
+        return $routes;
+    }
+
+    /**
+     * @param Application $application
+     * @return bool
+     */
+    protected function hasOctane(Application $application): bool
+    {
+        return $application?->octane !== null;
```
Comment (cosmastech) [inline src/Illuminate/Foundation/Console/RouteCacheCommand.php:108]: Isn't this adding null-safety to $application (which cannot be null or the interpreter will raise a TypeError) and not to the Application@$octane property?

Comment (CharlesBilbo) [reply src/Illuminate/Foundation/Console/RouteCacheCommand.php]: yeah it did i removed the method and changed it for the has method on the application context.

**Comment by CharlesBilbo** [comment]
i didnt add a test cause im not quite sure how yall would want this tested i could install octane since its technically a dependency to this PR but i wasnt quite sure on that. LMK and ill write the test if required.

File: src/Illuminate/Foundation/Console/RouteCacheCommand.php [hunk 2]
```diff
@@ -78,10 +79,24 @@ public function handle()
...
+        $application = $this->getFreshApplication();
+
+        $routes = tap($application['router']->getRoutes(), function ($routes) {
             $routes->refreshNameLookups();
             $routes->refreshActionLookups();
         });
+
+        if ($application->has('octane')) {
+            $warning = 'Running route:cache while running swoole decreases memory performance';
+            $this->components->warn($warning);
```
Comment (cosmastech) [inline src/Illuminate/Foundation/Console/RouteCacheCommand.php:91]: You could put this all onto one line, I'm not sure there's a good reason to store the warning as a string first.

**Comment by taylorotwell** [comment]
Octane::route is not really a documented stable feature of Octane at the current moment.

**Pull request closed without merging**